src/module.c

```diff
@@ -2305,7 +2305,7 @@ mstime_t RM_GetExpire(RedisModuleKey *key) {
  * The function returns REDISMODULE_OK on success or REDISMODULE_ERR if
  * the key was not open for writing or is an empty key. */
 int RM_SetExpire(RedisModuleKey *key, mstime_t expire) {
-    if (!(key->mode & REDISMODULE_WRITE) || key->value == NULL)
+    if (!(key->mode & REDISMODULE_WRITE) || key->value == NULL || expire < 0)
```

Comment:
`expire < 0` should be `expire <0 && expire != REDISMODULE_NO_EXPIRE`

or the else branch would be dead code.

The same with `RM_SetAbsExpire`.

Reply:
I have fixed it, thanks :)